bin/crystal

```diff
@@ -175,7 +175,7 @@ fi
 # CRYSTAL_PATH has all symlinks resolved. In order to avoid issues with duplicate file
 # paths when the working directory is a symlink, we cd into the current directory
 # with symlinks resolved as well (see https://github.com/crystal-lang/crystal/issues/12969).
-cd "$(realpath "$(pwd)")"
+cd "$(realpath "$(pwd)")" || exit
```

Comment:
What prompted this change? Do we fully understand it? Are there users who consistently have this line fail but work anyway? Why are we exiting with a successful status?

Reply:
I see, the page has all the answers: https://www.shellcheck.net/wiki/SC2164
And `exit` does not mean `exit 0` like I thought. Looks good then